addon/services/street.js

```diff
@@ -17,17 +27,76 @@ export default class StreetService extends GWRService {
     return this.cache(street);
   }
 
+  get primaryLanguage() {
+    return this.intl.primaryLocale?.split("-")?.[0];
+  }
+
   get language() {
-    return languageOptions[this.intl.primaryLocale?.split("-")?.[0]];
+    return this.languages.languageToCode(this.primaryLanguage);
   }
 
-  async search(query = {}) {
-    return super.search(query, query.EGID, {
+  @task
+  *_search(query) {
+    return yield super.search(query, query.EGID, {
       xmlMethod: "getStreet",
       urlPath: "streets",
       listModel: StreetList,
       listKey: "street",
       searchKey: "streetWithoutStreetGeometryType",
     });
   }
+
+  resetCachedLanguages() {
+    this.cachedLanguageOverrides = new Set([]);
+  }
+
+  async searchMultiple(query, useCachedLanguagesOnly) {
+    let results = [];
+
+    for (const lang of useCachedLanguagesOnly
+      ? this.cachedLanguageOverrides
+      : this.languages.availableLanguages) {
+      const code = this.languages.languageToCode(lang);
+      query.language = code;
+
+      results.push({
+        lang: [lang, code],
+        result: this._search.perform(query, query.EGID, {
+          xmlMethod: "getStreet",
+          urlPath: "streets",
+          listModel: StreetList,
+          listKey: "street",
+          searchKey: "streetWithoutStreetGeometryType",
+        }),
+      });
+    }
+
+    results = await Promise.allSettled(
+      results.map(async ({ lang, result }) => {
+        return { lang, result: await result };
+      }),
+    );
+    return results
+      .filter(({ value }) => value.result?.length)
+      .reduce((previous, { value: { lang, result } }) => {
+        this.cachedLanguageOverrides.add(lang[0]);
+        return previous.concat(result);
+      }, []);
+  }
+
+  async search(query = {}) {
+    const lastQueryString =
+      this._search.lastPerformed?.args[0]?.description?.descriptionLong.replaceAll(
+        "*",
+        "",
+      ) || "";
+    const currentQueryString =
+      query.description?.descriptionLong?.replaceAll("*", "") || "";
+
+    if (lastQueryString && currentQueryString.includes(lastQueryString)) {
```

Comment:
this might deserve a little comment explaining why we do this, something like "in order to avoid unnecessary requests while typing (and thereby "refining" the search), we continue searching only with the languages for which we previously received results."

Reply:
How do we avoid race conditions if `searchMultiple(query, true)` is called multiple times in a short timespan? shouldn't we maybe make sure that the language cache is only used if it's non-empty? maybe using ember concurrency for `searchMultiple` would also make sense?

Reply:
* add comment: `done`
* refactor to ec-task: `draft done`

Reply:
Looks great now! :+1: